src/main/java/net/mcreator/ui/minecraft/villagers/JVillagerTradeEntry.java

```diff
@@ -0,0 +1,154 @@
+	private final JSpinner priceMultiplier = new JSpinner(new SpinnerNumberModel(0.05, 0, 1, 0.01));
+
+	public JVillagerTradeEntry(MCreator mcreator, IHelpContext gui, JPanel parent, List<JVillagerTradeEntry> entryList) {
+		setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
+
+		setBackground(((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT")).darker());
+
+		price1 = new MCItemHolder(mcreator, ElementUtil::loadBlocksAndItems);
+		price2 = new MCItemHolder(mcreator, ElementUtil::loadBlocksAndItems);
+		sale = new MCItemHolder(mcreator, ElementUtil::loadBlocksAndItems);
+
+		final JComponent container = PanelUtils.expandHorizontally(this);
+
+		parent.add(container);
+		entryList.add(this);
+
+		priceMultiplier.setPreferredSize(new Dimension(92, 22));
+
+		JPanel line1 = new JPanel(new FlowLayout(FlowLayout.LEFT));
+		line1.setOpaque(false);
+
+		line1.add(L10N.label("elementgui.villager_trade.price1"));
+		line1.add(price1);
+		line1.add(L10N.label("elementgui.villager_trade.count_price_sale"));
+		line1.add(countPrice1);
+
+		line1.add(new JEmptyBox(15, 5));
+
+		line1.add(L10N.label("elementgui.villager_trade.price2"));
+		line1.add(price2);
+		line1.add(L10N.label("elementgui.villager_trade.count_price_sale"));
+		line1.add(countPrice2);
+
+		line1.add(new JEmptyBox(15, 5));
+
+		line1.add(L10N.label("elementgui.villager_trade.sale"));
+		line1.add(sale);
+		line1.add(L10N.label("elementgui.villager_trade.count_price_sale"));
+		line1.add(countSale);
+
+		JButton remove = new JButton(UIRES.get("16px.clear"));
+		remove.setText(L10N.t("elementgui.villager_trade.remove_entry"));
+		remove.addActionListener(e -> {
+			entryList.remove(this);
+			parent.remove(container);
+			parent.revalidate();
+			parent.repaint();
+		});
+
+		JPanel line2 = new JPanel(new FlowLayout(FlowLayout.LEFT));
+		line2.setOpaque(false);
+
+		line2.add(L10N.label("elementgui.villager_trade.level"));
+		line2.add(level);
+		line2.add(L10N.label("elementgui.villager_trade.max_trades"));
+		line2.add(maxTrades);
+		line2.add(L10N.label("elementgui.villager_trade.xp"));
+		line2.add(xp);
+		line2.add(L10N.label("elementgui.villager_trade.price_multiplier"));
+		line2.add(priceMultiplier);
+
+		add(PanelUtils.centerAndEastElement(line1, PanelUtils.join(remove)));
+		add(line2);
+
+		parent.revalidate();
+		parent.repaint();
+	}
+
+	public void reloadDataLists() {
+	}
+
+	public VillagerTrade.CustomTradeEntry.Entry getEntry() {
+		if (!price1.containsItem())
+			return null;
+		if (!sale.containsItem())
+			return null;
+
+		VillagerTrade.CustomTradeEntry.Entry entry = new VillagerTrade.CustomTradeEntry.Entry();
+		entry.price1 = price1.getBlock();
+		entry.countPrice1 = (int) countPrice1.getValue();
+		entry.price2 = price2.getBlock();
+		entry.countPrice2 = (int) countPrice2.getValue();
+		entry.sale = sale.getBlock();
+		entry.countSale = (int) countSale.getValue();
+		entry.level = level.getSelectedIndex() + 1;
```

Comment:
Are you sure in the code, levels don't start with 0 for novice and not 1?

Reply:
Yes I am sure! I'm using this system in my mod Electrona and it works fine.

Reply:
Please do not mark resolved without me checking it, asking for nth time now

Reply:
Ok